### ISSUE TITLE
Lint if a field overrides a superclass field

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/linter#1058]
E.g.
```dart
class C {
  var f;
}
class D extends C {
  var f; // LINT
}
class E implements C {
  var f; // OK
}
```

A lint warning me that I was overriding a superclass field would have saved me about two hours of debugging today :)

**Comment by a14n** [comment]
I just tested and this is catched by `overridden_fields` rule.

**Comment by pq** [comment]
@stereotype441 : was this in the analyzer code base?

Incidentally, I take the fact that there is a rule that would catch this to be another case in point for the "we should really have a set of lint rules defined so that projects can easily derive from them" conversation (#288).

**Comment by pq** [comment]
/cc @davidmorgan as this is a good Google ruleset consideration.

A topic for another issue elsewhere, but @stereotype441, @bwilkerson, it might be interesting to consider updating options for `front_end`, `kernel`, `analyzer` and `server` to at least derive from the evolving google ruleset.

But anyway, lets close this in favor of picking that up elsewhere if there's interest.

Thanks!

**Comment by davidmorgan** [comment]
Noted. Thanks!
